project-koku/koku
# [COST-3370] - drop ranking field from query data

## Jira Ticket

[COST-3370](https://issues.redhat.com/browse/COST-3370)

## Description

This change will ensure the ranking field is dropped from the original query data.
This query:
```
/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[request]=desc
```
would result in the  `ranks_by_day` and `data_frame` containing a `request` column. When these dataframes were merged, it resulted in a new DF with `request_x` and `request_y` fields. Further along in the query_handler, when ordering the data, we would try to sort by `request` but this field no longer existed in the data.

## Testing

1. Ingest OCP data.
2. Go to http://localhost:8000/api/cost-management/v1/reports/openshift/compute/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[request]=desc
3. see data.
4. Go to http://localhost:8000/api/cost-management/v1/reports/openshift/compute/?filter[limit]=50&filter[offset]=0&filter[resolution]=monthly&filter[time_scope_units]=month&filter[time_scope_value]=-2&group_by[project]=*&order_by[delta]=desc&delta=request__limit
5. see data.

## Notes

...
